### PR TITLE
tests: add DNS check to 10-network

### PR DESCRIPTION
**Pull request by sirspinach**
"The 13-hostnames test checks that DNS is working properly by
asserting that all containers can resolve the
hostnames of other containers (3.red.q), of individual labels
(blue.q), and of external resources (google.com)."

The test runs successfully when rebased on top of ethan's DNS pull request, but currently fails because external resources are not resolved correctly. Also, before the DNS pull request, a container with label A could not resolve container with label B unless A was connected to B.

Test output looks like this:
https://gist.github.com/sirspinach/413584dddcfa0d70ec94a69e8d6d0650

### CODE REVIEW

File: quilt-tester/tests/13-hostnames/check_hostnames.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/NetSys/quilt/api"
+	"github.com/NetSys/quilt/api/client/getter"
+	"github.com/NetSys/quilt/db"
+)
+
+// anyIPAllowed is used to indicate that any non-error response is okay for an external
+// query.
+const anyIPAllowed = "any ip is allowed"
```
Comment (kklin) [inline quilt-tester/tests/13-hostnames/check_hostnames.go]: Did you consider using a netmask to represent the expected IP? Then any is a `/0`, and Quilt IPs are a `/32`.

Comment (sirspinach) [inline quilt-tester/tests/13-hostnames/check_hostnames.go]: Hmm.. right now I think using a netmask is overkill. I don't want to have to wrap up all the ip strings inside structs just to remove this constant.

Comment (ejj) [inline quilt-tester/tests/13-hostnames/check_hostnames.go]: I'd at least make the constant 0.0.0.0, that's pretty universally known as any IP

Comment (sirspinach) [inline quilt-tester/tests/13-hostnames/check_hostnames.go]: changed anyIPAllowed to "0.0.0.0"

**Comment by sirspinach** [comment]
merged DNS test into 10-network/check_network.go.
Ready for further review.

File: quilt-tester/tests/10-network/check_network.go [hunk 2]
```diff
@@ -125,7 +143,9 @@ func runTests(tester networkTester, containers []db.Container) []testResult {
 type networkTester struct {
 	labelMap      map[string]db.Label
 	connectionMap map[string][]string
+	hostnameIPMap map[string]string
 	allIPs        []string
+	allHostnames  []string
```
Comment (kklin) [inline quilt-tester/tests/10-network/check_network.go]: Could we just iterate over the keys of `hostnameIPMap` where we need to? It'd be more direct.

**Comment by sirspinach** [comment]
removed `allHostnames`

**Comment by kklin** [comment]
@ejj looks like quilt-bot didn't assign this to you, but it's ready for you to review.

**Pull request merged**